extension/manifest.go

```diff
@@ -0,0 +1,77 @@
+package extension
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+const manifestFileName = "manifest.yaml"
+
+// Manifest contains information on the extension in local machine
+type Manifest struct {
+	Update    time.Time   `yaml:"update"`
+	Metadatas []*Metadata `yaml:"metadatas"`
+}
+
+// Metadata contains metadata for a particular extension
+type Metadata struct {
+	Owner     string   `yaml:"owner"`
+	Repo      string   `yaml:"repo"`
+	Aliases   []string `yaml:"aliases"`
+	Tag       string   `yaml:"tag"`
+	LocalPath string   `yaml:"local_path"`
+}
+
+// LoadManifest loads manifest from local machine
+func LoadManifest() (*Manifest, error) {
+	extensionDir, err := getExtensionDir()
+	if err != nil {
+		return nil, err
+	}
+	manifestPath := path.Join(extensionDir, manifestFileName)
+
+	manifest := &Manifest{}
+	if _, err := os.Stat(manifestPath); err == nil {
+		content, err := ioutil.ReadFile(manifestPath)
+		if err != nil {
+			return nil, fmt.Errorf("error reading file: %v", err)
+		}
+		err = yaml.Unmarshal(content, manifest)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshalling content: %v", err)
+		}
+	}
+	return manifest, nil
+}
+
+// FlushManifest flushes manifest into a file in local machine
+func FlushManifest(manifest *Manifest) error {
+	extensionDir, err := getExtensionDir()
+	if err != nil {
+		return err
+	}
+	content, err := yaml.Marshal(manifest)
+	if err != nil {
+		return fmt.Errorf("error marshalling manifest: %v", err)
+	}
+	if err := os.MkdirAll(extensionDir, os.ModePerm); err != nil {
+		return fmt.Errorf("error creating dir: %v", err)
+	}
+	manifestPath := path.Join(extensionDir, manifestFileName)
+	f, err := os.OpenFile(manifestPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0755)
+	if err != nil {
+		return fmt.Errorf("error opening file: %v", err)
+	}
+	defer f.Close()
+
+	_, err = f.Write(content)
+	if err != nil {
+		return fmt.Errorf("error writing file: %v", err)
+	}
+	return nil
+}
```

Comment:
Metadata contains information about an extension, such as the path where it is located in the user's local machine or alias when calling it from Optimus. With the metadata being known, it will be easy to install and to execute a particular extension. The user can install more than one extension in their local. And each extension requires metadata to be specified. And all of those metadata are stored in a file named Manifest.

Reply:
Yes, at least `alias` is required. And the way we store it, if not in the file, where would you suggest it would be?

Reply:
Also, later we planned to have an `Upgrade` function, which requires the repository information such as its ID and the current tag. Apart from that, though it's still in consideration, there might be a `List` function where it will list all the installed extensions and also check if there's a possible update those extensions.

Reply:
Alright, that make sense.